keps/sig-storage/3762-persistent-volume-last-phase-transition-time/README.md

```diff
@@ -0,0 +1,935 @@
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Unit tests completed and enabled
+- Add unit tests covering feature enablement/disablement.
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Allowing time for feedback (at least 2 releases between beta and GA).
+- Manually test version skew between the API server and KCM. See the expected
+  behavior below in Version Skew Strategy.
+- Manually test upgrade->downgrade->upgrade path.
+
+#### GA
+
+- No users complaining about the new behavior.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+No change in cluster upgrade / downgrade process.
+
+When upgrading the new `LastPhaseTransitionTime` field and its value will be added to a PV when it transitions phase.
+
+When downgrading from a version that added the new timestamp field PVs we need to make sure that after downgrade the
+values of the disabled field are removed. We intend to use API server strategy implementation, more specifically PV
+validation, to remove the values - each time a PV gets validated we will set the value to `nil` if feature gate is disabled.
```

Comment:
Actually one more thing - k8s validation isn't generally supposed to change anything.

I think what you really want is effectively add that to PrepareForCreate/PrepareForUpdate methods:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/persistentvolume/strategy.go

Reply:
Right, this is where I was aiming but shouldn't have used the term "validation" this way. I changed this to be specific about the strategy and the concrete methods.